tools/osbuilder/image-builder/image_builder.sh

```diff
@@ -372,6 +372,55 @@ create_disk() {
 	OK "Partitions created"
 }
 
+create_erofs_rootfs_image() {
+	local rootfs="$1"
+	local image="$2"
+	local fs_type="$3"
+	local block_size="$4"
+
+	if [ $block_size -ne 4096 ]; then
+		die "Invalid block size for erofs"
+	fi
+
+	readonly mount_dir=$(mktemp -p ${TMPDIR:-/tmp} -d osbuilder-mount-dir.XXXX)
+
+	info "Copying content from rootfs to root partition"
+	cp -a "${rootfs}"/* "${mount_dir}"
+
+	info "Removing unneeded systemd services and sockets"
+	for u in "${systemd_units[@]}"; do
+		find "${mount_dir}" -type f \( \
+			 -name "${u}.service" -o \
+			 -name "${u}.socket" \) \
+			 -exec rm -f {} \;
+	done
+
+	info "Removing unneeded systemd files"
+	for u in "${systemd_files[@]}"; do
+		find "${mount_dir}" -type f -name "${u}" -exec rm -f {} \;
+	done
+
+	info "Creating empty machine-id to allow systemd to bind-mount it"
+	touch "${mount_dir}/etc/machine-id"
+
+	readonly fsimage="$(mktemp)"
+	mkfs.erofs "${fsimage}" "${mount_dir}"
```

Comment:
should we use `-E force-inode-compact` ?

Reply:
If files are all < 4GB and no need to remain some timestamp, "-E force-inode-compact" is fine.
But I think it's no necessary to use it since mkfs.erofs uses compact inodes by default (except for the files >= 4GB or some cases in which extended inodes are really needed.)

Reply:
If files are all < 4GB and no need to remain some timestamp, "-E force-inode-compact" is fine.
But I think it's no necessary to use it since mkfs.erofs uses compact inodes by default (except for the files >= 4GB or some cases in which extended inodes are really needed.)